### ISSUE TITLE
Extend rule func-call-spacing to allow newlines with option "never"

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#7587]
I have a project where the function name and the left parenthesis in a calling expression are either attached without spaces or separated by a newline (followed by optional indent spaces).

By now, I'm still using JSCS with the rule `disallowSpacesInCallExpression` to enforce this style.

**Examples of correct code with the JSCS rule `disallowSpacesInCallExpression`**

```js
obj.doSomething("foo", "bar");

obj.doSomething
("foo", "bar");
```

**Examples of incorrect code with the JSCS rule `disallowSpacesInCallExpression`**

```js
obj.doSomething ("foo", "bar");
```

Unfortunately, I haven't managed to port this rule to ESLint yet.
`func-call-spacing: ["error", "never"]` disallows both white spaces and newlines.

What I would love to accomplish with ESLint is a config to allow only newlines or no spacing at all between the function name and the left parenthesis in a calling expression.

I was thinking of something like this:

```js
func-call-spacing: ["error", "never", { "allowNewlines": true }]
```

A new option could be more flexible, though.

If this is not feasible, I think I'm going to write my own custom rule so that I can get rid of the
JSCS dependency.

**Comment by not-an-aardvark** [comment]
Thanks for the suggestion.

In my opinion, this type of code style is a bit confusing:

```js
obj.doSomething
("foo", "bar");
```

But I wouldn't be opposed to adding an option for it if the rest of the team is in favor if it.

**Comment by vitorbal** [comment]
I've never seen that coding style before. Are there any known codebases out there using this? I'm asking this because I'd doubt we would be able to justify a new option to `func-call-spacing` If it's just to support a niche use case.

**Comment by vitorbal** [comment]
Ah, just re-read this and noticed you mentioned the rule comes from JSCS. In that case, I think this needs more careful consideration. Not sure if allowing newlines in `disallowSpacesInCallExpression` was intentional or a bug. @eslint/eslint-team maybe someone from the JSCS team could help clarify?

**Comment by fasttime** [comment]
Hmm... yes, that was a bit confusing. In fact my code looks more like this:

```js
myObj =
{
    someProperty: someFunctionWithManyArgs
    (
        "foo",
        "bar",
        // ...
        "baz"
    )
};
```

The option to allow newlines is already there, but it's bundled with `func-call-spacing: ["error", "always"]`, so at least it looks like I'm not the only one to use this kind of style. But please, don't ask me what codebases out there are using this 😐

**Comment by fasttime** [comment]
I've now created [a plugin](https://github.com/fasttime/eslint-plugin-no-spaces-in-call-expression) that fits my (apparently quite special) requirements. Anyway, after taking a closer look at the source of `func-call-spacing`, I think it wouldn't be hard at all to extend the core rule in order to provide more flexibility/JSCS compatibility. Well, unless I'm overlooking some obvious complications.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
